src/Shmuelie.WinRTServer/ComServer.cs

```diff
@@ -189,6 +189,9 @@ private void Factory_InstanceCreated(object? sender, InstanceCreatedEventArgs e)
         liveServers.AddLast(new WeakReference(e.Instance));
         InstanceCreated?.Invoke(this, e);
         firstInstanceCreated?.TrySetResult(e.Instance);
+
+        // Reset the TaskCompletionSource to avoid keeping a reference to the first instance created indefinitely.
+        firstInstanceCreated = new TaskCompletionSource<object>();
```

Comment:
I can also add a bool property to keep track whether this "event" was raised already and skip `trySetResult` if yes.
I'm not sure if it's needed cuz users will usually `await` this event only once so `trySetResult` on new instance of it yields no effect/change in behavior.